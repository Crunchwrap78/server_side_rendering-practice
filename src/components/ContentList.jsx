import React, { Component } from 'react';

export default class ContentList extends Component {
  constructor () {
    super()
    this.listContent = this.listContent.bind(this)
  }

  listContent() {
    const { listingsData } = this.props;

    if (listingsData == undefined || listingsData.length == 0) {
      return "Sorry your filter did not match any listing"
    }

    return listingsData.map((listing, index) => {
      if(this.props.globalState.view == 'box') {
        return (<div className="col-md-4" key={index}>
          <div className="listing">
            <div className="listing-img" style={{background: `url("${listing.image}") no-repeat center center`}}>
              <span className="address">{listing.address}</span>
              <div className="details">
                <div className="col-md-3">
                  <div className="user-img"> </div>
                </div>

                <div className="col-md-9">
                  <div className="user-details">
                    <span className="user-name">Nina Smith</span>
                    <span className="post-date">05/05/2017</span>
                  </div>
                  <div className="listing-details">
                    <div className="floor-space">
                      <i className="fa fa-square-o" aria-hidden="true"></i>
                      <span>{listing.floorSpace} ft&sup2;</span>
                    </div>
                    <div className="bedrooms">
                      <i className="fa fa-bed" aria-hidden="true"></i>
                      <span>{listing.rooms} bedrooms</span>
                    </div>
                  </div>
                  <div className="view-btn">
                    View Listing
                  </div>
                </div>
              </div>
            </div>
            <div className="bottom-info">
              <span className="price">${listing.price}</span>
              <span className="location"><i className="fa fa-map-marker" aria-hidden="true"></i> {listing.city}, {listing.state} </span>
            </div>
          </div>
        </div>)
      } else {
        // THIS IS LONG VIEW
        return (
          <div className="col-md-12 col-lg-6" key={index}>
            <div className="listing">
              <div className="listing-img" style={{background: `url("${listing.image}") no-repeat center center`}}>
                <span className="address">{listing.address}</span>
                <div className="details">
                  <div className="col-md-3">
                    <div className="user-img"> </div>
                  </div>
                  <div className="col-md-9">
                    <div className="user-details">
                      <span className="user-name">Nina Smith</span>
                      <span className="post-date">05/05/2017</span>
                    </div>
                    <div className="listing-details">
                      <div className="floor-space">
                        <i className="fa fa-square-o" aria-hidden="true"></i>
                        <span>{listing.floorSpace} ft&sup2;</span>
                      </div>
                      <div className="bedrooms">
                        <i className="fa fa-bed" aria-hidden="true"></i>
                        <span>{listing.rooms} bedrooms</span>
                      </div>
                    </div>
                    <div className="view-btn">
                      View Listing
                    </div>
                  </div>
                </div>
              </div>
              <div className="bottom-info">
                <span className="price">${listing.price}</span>
                <span className="location"><i className="fa fa-map-marker" aria-hidden="true"></i> {listing.city}, {listing.state} </span>
              </div>
            </div>
          </div>
          )
        }
      })
    }

    render() {
      return (
        <section id="content-list">
          <section className="search-area" >
            <input type="text" name="search" onChange={this.props.handleChange}/>
          </section>

          <section className="sortby-area">
            <div className="results">{this.props.globalState.filteredData.length} results found</div>
            <div className="sort-options">
              <select name="sortby" className="sortby" onChange={this.props.handleChange}>
                <option value="price-asc">Highest Price</option>
                <option value="">Lowest Price</option>
              </select>
              <div className="view">
                <i className="fa fa-th-list" aria-hidden="true" onClick={()=>this.props.changeView("long")}></i>
                <i className="fa fa-th" aria-hidden="true" onClick={()=>this.props.changeView("box")}></i>
              </div>
            </div>
          </section>

          <section className="listings-results">
            {this.listContent()}
          </section>

          <section id="pagination">
            <ul className="pages">
              <li>Prev</li>
              <li className="active">1</li>
              <li>2</li>
              <li>3</li>
              <li>4</li>
              <li>5</li>
              <li>Next</li>
            </ul>
          </section>
        </section>
      )
    }
}
