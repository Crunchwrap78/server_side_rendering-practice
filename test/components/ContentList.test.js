import React from 'react';
import { shallow } from 'enzyme';
import ContentList from '../../src/components/ContentList';
import listingsData from '../../src/data/listings.js'

let handleChange, changeView, wrapper;
beforeEach(() => {
  const globalState = {
    city: 'All',
    rooms: 0,
    listingsData,
    homeType: 'All',
    min_price: 0,
    max_price: 10000000,
    min_floor_space: 0,
    max_floor_space: 4000,
    elevator: false,
    basement: false,
    swimming_pool: false,
    gym: false,
    filteredData: listingsData,
    populateDropDown: '',
    sortby: 'price-dsc',
    search: '',
    view: 'long'
  };
  handleChange = jest.fn();
  changeView = jest.fn();
  wrapper = shallow(
    <ContentList
      handleChange={handleChange}
      changeView={changeView}
      globalState={globalState}
    />
  );
});

test('should render ContentList', () => {
  expect(wrapper).toMatchSnapshot();
});


test('should handle search text when input changes', () => {
  const value = 'ooo';
  wrapper.find('input').simulate('change', {
    target: { value }
  });
  expect(handleChange).toHaveBeenCalled();
});

test('should handle view change', () => {
  const value = 'box';
  wrapper.find('i').at(1).simulate('click');
  expect(changeView).toHaveBeenLastCalledWith(value);
});
