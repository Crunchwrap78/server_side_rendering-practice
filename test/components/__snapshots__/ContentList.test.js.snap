// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ContentList 1`] = `
<section
  id="content-list"
>
  <section
    className="search-area"
  >
    <input
      name="search"
      onChange={[Function]}
      type="text"
    />
  </section>
  <section
    className="sortby-area"
  >
    <div
      className="results"
    >
      7
       results found
    </div>
    <div
      className="sort-options"
    >
      <select
        className="sortby"
        name="sortby"
        onChange={[Function]}
      >
        <option
          value="price-asc"
        >
          Highest Price
        </option>
        <option
          value=""
        >
          Lowest Price
        </option>
      </select>
      <div
        className="view"
      >
        <i
          aria-hidden="true"
          className="fa fa-th-list"
          onClick={[Function]}
        />
        <i
          aria-hidden="true"
          className="fa fa-th"
          onClick={[Function]}
        />
      </div>
    </div>
  </section>
  <section
    className="listings-results"
  >
    Sorry your filter did not match any listing
  </section>
  <section
    id="pagination"
  >
    <ul
      className="pages"
    >
      <li>
        Prev
      </li>
      <li
        className="active"
      >
        1
      </li>
      <li>
        2
      </li>
      <li>
        3
      </li>
      <li>
        4
      </li>
      <li>
        5
      </li>
      <li>
        Next
      </li>
    </ul>
  </section>
</section>
`;
